README.md

```diff
@@ -489,6 +532,42 @@ p := Exec("ls /var/app/config/").Concat().Stdout()
 
 Each input file will be closed once it has been fully read.
 
+## Dirname
+
+`Dirname()` reads a list of filenames from the pipe, one per line, and returns a pipe which contains only the directory names of each filename.
+
+If a line is empty, `Dirname()` will return a '.' for that line
```

Comment:
This is how `dirname(1)` behaves in UNIX shell scripts. It returns a '.' because typically folks write code like this:

```
MY_PATH="$(dirname $foo)/my_file.ext"

If it didn't return the '.', then MY_PATH would end up as `/my_file.ext`.

Reply:
Hmm. I think we need to run up a use case here which demonstrates this point; a shell script analogy isn't necessarily persuasive, because `script` programs aren't much like shell scripts, except in function.